### ISSUE TITLE
Add configuration option to ignore increments on strings

### ISSUE THREAD

**Issue by kelunik** [vimeo/psalm#2061]
```php
$nextId = 'a';

return $nextId++;
```

Code like this isn't used often, but if libraries make use of it, there are usually many cases that are reported as errors. It probably makes sense to leave it an error by default, but offer a configuration option to optionally ignore increments on strings being an error.

See also #1944.

**Comment by muglug** [comment]
I've made it a separate issue type you can suppress (`StringIncrement`).

**Comment by kelunik** [comment]
Thanks!
